vault/activity_log_testing_util.go

```diff
@@ -0,0 +1,167 @@
+func (a *ActivityLog) GetCurrentEntities() *activity.EntityActivityLog {
+	a.l.RLock()
+	defer a.l.RUnlock()
+	return a.currentSegment.currentEntities
+}
+
+// WriteToStorage is used to put entity data in storage
+// `path` should be the complete path (not relative to the view)
+func WriteToStorage(t *testing.T, c *Core, path string, data []byte) {
+	t.Helper()
+	err := c.barrier.Put(context.Background(), &logical.StorageEntry{
+		Key:   path,
+		Value: data,
+	})
+
+	if err != nil {
+		t.Fatalf("Failed to write %s\nto %s\nerror: %v", data, path, err)
+	}
+}
+
+// SetStandbyEnable sets enabled on a performance standby (using config)
+func (a *ActivityLog) SetStandbyEnable(ctx context.Context, enabled bool) {
+	var enableStr string
+	if enabled {
+		enableStr = "enable"
+	} else {
+		enableStr = "disable"
+	}
+
+	// TODO only patch enabled?
+	a.SetConfigStandby(ctx, activityConfig{
+		DefaultReportMonths: 12,
+		RetentionMonths:     24,
+		Enabled:             enableStr,
+	})
+}
+
+// ExpectCurrentSegmentRefreshed verifies that the current segment has been refreshed
+// non-nil empty components and updated with the `expectedStart` timestamp
+// Note: if `verifyTimeNotZero` is true, ignore `expectedStart` and just make sure the timestamp isn't 0
+func (a *ActivityLog) ExpectCurrentSegmentRefreshed(t *testing.T, expectedStart int64, verifyTimeNotZero bool) {
+	t.Helper()
+
+	a.l.RLock()
+	defer a.l.RUnlock()
+	a.fragmentLock.RLock()
+	defer a.fragmentLock.RUnlock()
+	if a.currentSegment.currentEntities == nil {
+		t.Fatalf("expected non-nil currentSegment.currentEntities")
+	}
+	if a.currentSegment.currentEntities.Entities == nil {
+		t.Errorf("expected non-nil currentSegment.currentEntities.Entities")
+	}
+	if a.activeEntities == nil {
+		t.Errorf("expected non-nil activeEntities")
+	}
+	if a.currentSegment.tokenCount == nil {
+		t.Fatalf("expected non-nil currentSegment.tokenCount")
+	}
+	if a.currentSegment.tokenCount.CountByNamespaceID == nil {
+		t.Errorf("expected non-nil currentSegment.tokenCount.CountByNamespaceID")
+	}
+
+	if len(a.currentSegment.currentEntities.Entities) > 0 {
+		t.Errorf("expected no current entity segment to be loaded. got: %v", a.currentSegment.currentEntities)
+	}
+	if len(a.activeEntities) > 0 {
+		t.Errorf("expected no active entity segment to be loaded. got: %v", a.activeEntities)
+	}
+	if len(a.currentSegment.tokenCount.CountByNamespaceID) > 0 {
+		t.Errorf("expected no token counts to be loaded. got: %v", a.currentSegment.tokenCount.CountByNamespaceID)
+	}
+
+	if verifyTimeNotZero {
+		if a.currentSegment.startTimestamp == 0 {
+			t.Error("bad start timestamp. expected no reset but timestamp was reset")
+		}
+	} else if a.currentSegment.startTimestamp != expectedStart {
+		t.Errorf("bad start timestamp. expected: %v got: %v", expectedStart, a.currentSegment.startTimestamp)
+	}
+}
+
+// ActiveEntitiesEqual checks that only the set of `test` exists in `active`
```

Comment:
Let's put this in testhelpers too.  The `vault` package is already enormous and there's no reason this needs to be here that I can see.

Reply:
same problem as above (import cycle). Is there somewhere else we could put this?